packages/react-devtools-scheduling-profiler/src/content-views/UserTimingMarksView.js

```diff
@@ -0,0 +1,230 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {UserTimingMark} from '../types';
+import type {Interaction, MouseMoveInteraction, Rect, Size} from '../view-base';
+
+import {
+  positioningScaleFactor,
+  timestampToPosition,
+  positionToTimestamp,
+  widthToDuration,
+} from './utils/positioning';
+import {
+  View,
+  Surface,
+  rectContainsPoint,
+  rectIntersectsRect,
+  intersectionOfRects,
+} from '../view-base';
+import {
+  COLORS,
+  EVENT_ROW_PADDING,
+  EVENT_DIAMETER,
+  BORDER_SIZE,
+} from './constants';
+
+const ROW_HEIGHT_FIXED = EVENT_ROW_PADDING + EVENT_DIAMETER + EVENT_ROW_PADDING;
+
+export class UserTimingMarksView extends View {
+  _marks: UserTimingMark[];
+  _intrinsicSize: Size;
+
+  _hoveredMark: UserTimingMark | null = null;
+  onHover: ((mark: UserTimingMark | null) => void) | null = null;
+
+  constructor(
+    surface: Surface,
+    frame: Rect,
+    marks: UserTimingMark[],
+    duration: number,
+  ) {
+    super(surface, frame);
+    this._marks = marks;
+
+    this._intrinsicSize = {
+      width: duration,
+      height: ROW_HEIGHT_FIXED,
+    };
+  }
+
+  desiredSize() {
+    return this._intrinsicSize;
+  }
+
+  setHoveredMark(hoveredMark: UserTimingMark | null) {
+    if (this._hoveredMark === hoveredMark) {
+      return;
+    }
+    this._hoveredMark = hoveredMark;
+    this.setNeedsDisplay();
+  }
+
+  /**
+   * Draw a single `UserTimingMark` as a circle in the canvas.
+   */
+  _drawSingleMark(
+    context: CanvasRenderingContext2D,
+    rect: Rect,
+    mark: UserTimingMark,
+    baseY: number,
+    scaleFactor: number,
+    showHoverHighlight: boolean,
+  ) {
+    const {frame} = this;
+    const {timestamp} = mark;
+
+    const x = timestampToPosition(timestamp, scaleFactor, frame);
+    const radius = EVENT_DIAMETER / 2;
+    const markRect: Rect = {
+      origin: {
+        x: x - radius,
+        y: baseY,
+      },
+      size: {width: EVENT_DIAMETER, height: EVENT_DIAMETER},
+    };
+    if (!rectIntersectsRect(markRect, rect)) {
+      return; // Not in view
+    }
+
+    const fillStyle = showHoverHighlight
+      ? COLORS.USER_TIMING_HOVER
+      : COLORS.USER_TIMING;
+
+    if (fillStyle !== null) {
+      const y = markRect.origin.y + radius;
+
+      context.beginPath();
+      context.fillStyle = fillStyle;
+      context.arc(x, y, radius, 0, 2 * Math.PI);
+      context.fill();
+    }
+  }
```

Comment:
Can we give these marks a different visual style, to set them apart from the e.g. state updates and suspends? Maybe something like:
```suggestion
  _drawSingleMark(
    context: CanvasRenderingContext2D,
    rect: Rect,
    mark: UserTimingMark,
    baseY: number,
    scaleFactor: number,
    showHoverHighlight: boolean,
  ) {
    const {frame} = this;
    const {timestamp} = mark;

    const x = timestampToPosition(timestamp, scaleFactor, frame);
    const size = USER_TIMING_MARK_SIZE;
    const halfSize = size / 2;

    const markRect: Rect = {
      origin: {
        x: x - halfSize,
        y: baseY,
      },
      size: {width: size, height: size},
    };
    if (!rectIntersectsRect(markRect, rect)) {
      return; // Not in view
    }

    const fillStyle = showHoverHighlight
      ? COLORS.USER_TIMING_HOVER
      : COLORS.USER_TIMING;

    if (fillStyle !== null) {
      const y = baseY + halfSize;

      context.beginPath();
      context.fillStyle = fillStyle;
      context.moveTo(x, y - halfSize);
      context.lineTo(x + halfSize, y);
      context.lineTo(x, y + halfSize);
      context.lineTo(x - halfSize, y);
      context.fill();
    }
  }
```

The above proposal also involves adding a new constant to `content-views/constants.js`:
```js
export const USER_TIMING_MARK_SIZE = 8;
```